src/object/srv_obj.c

```diff
@@ -2559,7 +2559,7 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	}
 
 again2:
-	if (orw->orw_iod_array.oia_oiods != NULL && split_req == NULL && tgt_cnt != 0) {
+	if (orw->orw_iod_array.oia_oiods != NULL && split_req == NULL) {
 		rc = obj_ec_rw_req_split(orw->orw_oid, &orw->orw_iod_array,
 					 orw->orw_nr, orw->orw_start_shard,
 					 orw->orw_tgt_max, PO_COMP_ID_ALL,
```

Comment:
probably you can check orw->orw_shard_tgts.ca_count instead?  can it be 0? otherwise worth Assertion?

Reply:
orw->orw_iod_array.oia_oiods non NULL then it must be EC, EC update must with orw_shard_tgts.ca_count >= 1, yes can assert.